media_driver/linux/common/os/mos_utilities_specific.c

```diff
@@ -2692,7 +2692,7 @@ uint32_t MOS_GetThreadId(
 
 uint32_t MOS_GetCurrentThreadId()
 {
-    return (uint32_t)pthread_self();
+    return (uintptr_t)pthread_self();
```

Comment:
Here we should use `_lwp_self()` for NetBSD and `uint64_t` to be more portable.

Reply:
> Here we should use `_lwp_self()` for NetBSD

Wouldn't be consistent with Linux as `pthread_t` is not an integer representation of thread identifier. If the implementation doesn't make sense then it's a separate bug, out of scope for this PR.

<details><summary>pthread_self vs. gettid</summary>

```c
$ cc test.c -o test -lpthread
$ ./test
34366243136
45912

$ cat test.c
#include <pthread.h>
#include <stdint.h>
#include <stdio.h>

#if defined(__BIONIC__)
#include <unistd.h>
#elif defined(__linux__)
#include <sys/syscall.h>
#elif defined(__DragonFly__) || defined(__FreeBSD__)
#include <pthread_np.h>
#elif defined(__NetBSD__)
#include <lwp.h>
#elif defined(__sun)
#include <thread.h>
#endif

#if !defined(__BIONIC__)
static pid_t gettid()
{
#if defined(__linux__)
  return syscall(__NR_gettid);
#elif defined(__DragonFly__) || defined(__FreeBSD__)
  return pthread_getthreadid_np();
#elif defined(__NetBSD__)
  return _lwp_self();
#elif defined(__OpenBSD__)
  return getthrid();
#elif defined(__sun)
  return thr_self();
#endif
}
#endif

int main()
{
  printf("%jd\n", (uintmax_t)pthread_self());
  printf("%jd\n", (uintmax_t)gettid());
}
```
</details>

> `uint64_t` to be more portable.

`uintptr_t` is already used in a number of places. media-driver only supports Intel architectures, so `uintptr_t` is either `uint32_t` or `uint64_t`.  On exotic platforms `uintptr_t` maybe smaller than `uint32_t` or larger than `uint64_t`. Even then using `uintptr_t` is likely more correct.

NetBSD i386 doesn't like `uint64_t` anyway:
```c
media_driver/linux/common/os/mos_utilities_specific.c: In function 'MOS_GetCurrentThreadId':
media_driver/linux/common/os/mos_utilities_specific.c:2695:12: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
     return (uint64_t)pthread_self();
            ^
```

Reply:
I mean `uint32_t MOS_GetCurrentThreadId()` should be `uint64_t MOS_GetCurrentThreadId()`.. but if we really need a pointer to `pthread_t` here as ID, it is good to go for `uintptr_t` here.

Reply:
Out of scope. Submit a PR yourself. I'm not interested in fixing an API that's not even used for anything important.

Reply:
OK. This patch is not making things worse than they already are.